Revit_Core_Engine/Create/AddInsulation.cs

```diff
@@ -0,0 +1,108 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using Autodesk.Revit.DB.Structure;
+using BH.Engine.Adapters.Revit;
+using BH.oM.Adapters.Revit.Settings;
+using BH.oM.Reflection.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using Autodesk.Revit.DB.Mechanical;
+using Autodesk.Revit.DB.Plumbing;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Add or update existing insulation of the duct or pipe with given inuslation type and thickness.")]
+        [Input("host", "Host element to add the insulation on.")]
+        [Input("insulationType", "Type of insulation to be added on the host object.")]
+        [Input("insulationThickness", "Thickness of insulation to be added on the host object in millimeters.")]
+        [Output("insulation", "Insulation that has been added on the host.")]
+        public static InsulationLiningBase AddInsulation(this Element host, ElementType insulationType, double insulationThickness)
```

Comment:
If this is `Create` method, I would simply add it `Insulation`. Alternatively, I would consider moving this method to `Modify` - actually this could make more sense because this does not always create a new insulation, it can also update the existing one?